include/boost/geometry/algorithms/detail/is_valid/polyhedral_surface.hpp

```diff
@@ -0,0 +1,501 @@
+// Boost.Geometry
+
+// Copyright (c) 2025 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+
+#include <boost/qvm/vec.hpp>
+#include <boost/qvm/vec_operations.hpp>
+
+#include <boost/geometry/algorithms/detail/is_valid/interface.hpp>
+#include <boost/geometry/algorithms/detail/is_valid/polygon.hpp>
+#include <boost/geometry/algorithms/detail/calculate_point_order.hpp>
+#include <boost/geometry/algorithms/correct.hpp>
+#include <boost/geometry/io/wkt/wkt.hpp>
+#include <boost/geometry/strategies/transform/matrix_transformers.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace is_valid
+{
+
+struct polygon_frontier_type : public std::stack<size_t>
+{
+    polygon_frontier_type(size_t size)
+        : m_stack(), m_visited(size, false)
+    {}
+
+    void push(size_t index)
+    {
+        if (!m_visited[index])
+        {
+            m_stack.push(index);
+            m_visited[index] = true;
+        }
+    }
+    bool empty()
+    {
+        return m_stack.empty();
+    }
+    size_t top()
+    {
+        return m_stack.top();
+    }
+    void pop()
+    {
+        m_stack.pop();
+    }
+private :
+    std::stack<size_t> m_stack;
+    std::vector<bool> m_visited;
+};
+
+template <typename PolyhedralSurface>
+struct valid_intersection
+{
+    valid_intersection(size_t num_of_polygons)
+        : m_polygon_projections(num_of_polygons)
+    {}
+
+    using coordinate_type = typename coordinate_type<PolyhedralSurface>::type;
+    using vec3 = boost::qvm::vec<coordinate_type, 3>;
+    using polygon_3d = typename boost::range_value<PolyhedralSurface>::type;
+    using ring_3d = typename boost::geometry::ring_type<polygon_3d>::type;
+    using point_3d = typename boost::geometry::point_type<ring_3d>::type;
+    using segment_3d = typename boost::geometry::model::segment<point_3d>;
+    using set_vec_from_point = typename strategy::transform::detail::matrix_transformer::
```

Comment:
The name is not 100% clear to me.

But (I see now) we have that already for a long time.
It sets a vector from a point?
Anyway, the `using` clause is fine then...